vyper_contracts/src/EIP3074/AuthorizedTransfer.vy

```diff
@@ -0,0 +1,14 @@
+# pragma version ~=0.4.0rc2
+
+from ethereum.ercs import IERC20
+
+# counter to keep track of the number of invokes
+counter: public(uint256)
+
+# authorise this contract to spend the tokens on behalf of the user
+@external
+def authorizeInvoke(sig: Bytes[97], token: IERC20, receiver: address, amount: uint256):
+    self.counter += 1
+    authorize(self, sig)
```

Comment:
I don't think authorize should take self as an argument? The address should be the one of the authority

Reply:
perhaps add a `sender` address function parameter that we will auth with